plugins/inputs/spectrum_scale/spectrum_scale.go

```diff
@@ -0,0 +1,416 @@
+package spectrum_scale
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Spectrum_Scale struct {
+	Sensors        []string
+	SocketLocation string
+}
+
+var sampleConfig = `
+  # An array of Spectrum Scale (GPFS) sensors
+  #
+  # These will be monitored through the mmpmon socket
+
+  sensors = ["nsd_ds", "gfis", "fis"]
+  socketLocation = "/var/mmfs/mmpmon/mmpmonSocket"
+
+`
+
+func (s *Spectrum_Scale) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Spectrum_Scale) Description() string {
+	return "Read metrics of Spectrum Scale"
+}
+
+// Reads stats from all configured servers accumulates stats.
+// Returns one of the errors encountered while gather stats (if any).
+func (s *Spectrum_Scale) Gather(acc telegraf.Accumulator) error {
+	var sensors []string
+	var socketLocation string
+
+	sensors = s.Sensors
+	socketLocation = s.SocketLocation
+
+	if len(sensors) == 0 {
+		sensors = []string{"nsd_ds", "gfis", "fis"}
+	}
+
+	if len(socketLocation) == 0 {
+		socketLocation = "/var/mmfs/mmpmon/mmpmonSocket"
+	}
+
+	// For every sensor specified, connect to the socket and parse the result
+	for _, sensor := range sensors {
+		err := s.getSensor(socketLocation, sensor, acc)
```

Comment:
I'm not familiar with spectrums socket api so this might be a dumb question, but is there a way to get more than one sensor at once?  Iterating over all the sensors might mean the data returned for each sensor isn't from the exact same time which could be confusing.